src/main/java/ch/njol/skript/effects/EffDestroyableKeys.java

```diff
@@ -0,0 +1,85 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class EffDestroyableKeys extends Effect {
+
+	static {
+		Skript.registerEffect(EffDestroyableKeys.class,
+			"allow %~itemtypes% to (destroy|break|mine) %itemtypes/namespacedkeys%",
+			"prevent %~itemtypes% from (destroy|break|mine)[ing] %itemtypes/namespacedkeys%");
```

Comment:
Don't think Skript has a type called `namespacedkeys`, unless you want to add it in this pr?
This extends to the other classes too.
```suggestion
			"allow %~itemtypes% to (destroy|break|mine) %itemtypes%",
			"prevent %~itemtypes% from (destroy|break|mine)[ing] %itemtypes%");
```

Reply:
Would this be something worth adding to Skript for this pr? Happy to add it if you think it would be useful.

Reply:
I don't think Skript needs a namespaced key type, personally, but I haven't put a ton of thought into pros and cons. My gut just says "just use strings"

Reply:
I'm happy to just have these classes only work with itemtypes for now, and if some conversation in the future ends up in favour of adding the namespacedkeys then add them then?